csrc/scheduler/reduction.cpp

```diff
@@ -367,14 +367,14 @@ std::unique_ptr<ReductionParams> inner2dReductionHeuristic(
     rparams->grid_dim_iter_dom = ParallelType::BIDy;
     if (godim > scheduler_utils::y_grid_limit) {
       rparams->split_grid_dim_iter_dom_outer = true;
-      gdimy = std::min(godim, scheduler_utils::y_grid_limit);
+      gdimy = scheduler_utils::y_grid_limit;
     }
 
   } else {
     rparams->grid_dim_iter_dom = ParallelType::BIDx;
-    if (gdimx > scheduler_utils::x_grid_limit) {
+    if (godim > scheduler_utils::x_grid_limit) {
       rparams->split_grid_dim_iter_dom_outer = true;
-      gdimx = godim;
+      gdimx = scheduler_utils::x_grid_limit;
```

Comment:
cc'ing @liqiangxl , I don't quite follow the scheduler here. Are we mapping `gdimx` on outer dimension here?

Reply:
correct. The fix doesn't seem right. Should be
```
if (rparams->cross_grid_inner_reduction) {
   gdimx = grdim
   gdimy = godim
   further split if gdimy is too big
}else{
  gdimx = godim
  further split if gdimx is too big
}

```

Reply:
`FusionAvoidRedundantWrite_CUDA` complains when `gdimx = godim`, so I left it uninitialized.

Reply:
It is originally to be `gdimx = std::min(godim, scheduler_utils::x_grid_limit)`.
Since `gdimx` is only set in the if-statement, so `gdimx = scheduler_utils::x_grid_limit`.

Reply:
Make sense. since `gdimx` is directly used in `lparams`, we should leave it uninitialized when `rparams->split_grid_dim_iter_dom_outer = false;`. 
The scheduler parallelize the iteration dim with gdimx: e.g `blockIdx.x{i0}`, when `gdimx` is not set, its value will be determined at runtime.

 when extent of {i0} is larger than `x_grid_limit`, it further split {i0} to `{i0/x_grid_limit}, {x_grid_limit}`, and parallelize as `Serial{i0/x_grid_limit}, blockIdx.x{x_grid_limit}`, here  bdimx is fixed.